File: Argus/Angular-Frontend/src/app/model/notification.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {environment} from '../../environments/environment';
// import {environment} from '../../environments/environment.prod';

@Injectable({
  providedIn: 'root'
})
export class NotificationService {
  private baseUrl = `${environment.apiUrl}/springboot/api/notifications`;

  constructor(private http: HttpClient) { }

  getNotificationById(id: number): Observable<any> {
    return this.http.get(`${this.baseUrl}/${id}`);
  }

  addNotification(notification: any): Observable<any> {
    return this.http.post(`${this.baseUrl}`, notification);
  }

  updateNotification(id: number, value: any): Observable<any> {
    return this.http.put(`${this.baseUrl}/${id}`, value);
  }

  deleteNotification(id: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/${id}`, { responseType: 'text' });
  }

  getNotificationList(): Observable<any> {
    return this.http.get(`${this.baseUrl}`);
  }
}

File: Spring-Backend/src/main/java/com/springboot/SpringBackend/service/NotificationService.java
package com.springboot.SpringBackend.service;

import com.springboot.SpringBackend.model.Notification;

import java.util.List;
import java.util.Optional;

public interface NotificationService {
    List<Notification> getAllNotifications();

    Optional<Notification> getNotificationById(Long id);

    Notification createNotification(Notification x);

    Notification updateNotification(Notification x);

    void deleteNotification(Notification x);

    void deleteAllNotifications();
}
